lemur/sources/service.py

```diff
@@ -184,6 +182,59 @@ def sync_endpoints(source):
     return new, updated, updated_by_hash
 
 
+def get_cert_for_endpoint(source, endpoint, certificate_name):
```

Comment:
This function contains existing functionality that was re-factored from `sync_endpoints()`.